OpenRA.Game/Graphics/WorldRenderer.cs

```diff
@@ -279,9 +332,12 @@ public void DrawAnnotations()
 
 				foreach (var b in World.ScreenMap.MouseBounds(World.RenderPlayer))
 				{
-					var points = b.Vertices
-						.Select(p => Viewport.WorldToViewPx(p).ToFloat2())
-						.ToArray();
+					var points = new float2[b.Vertices.Length];
```

Comment:
In practice, `b.Vertices.Length` should generally either be 4 or 6. Maybe not worth the bookkeeping overhead since this is for debug, but we could otherwise in principle reuse the same scratch array between loops and only reallocate it if the length has changed?